doc/api_assets/api.js

```diff
@@ -0,0 +1,131 @@
+        themeToggleButton.addEventListener('click', function() {
+          mq.removeEventListener('change', mqChangeListener);
+        }, { once: true });
+      }
+    }
+
+    if (mq.matches) {
+      document.documentElement.classList.add('dark-mode');
+    }
+  } else if (userSettings === 'true') {
+    document.documentElement.classList.add('dark-mode');
+  }
+
+  if (themeToggleButton) {
+    themeToggleButton.hidden = false;
+    themeToggleButton.addEventListener('click', function() {
+      sessionStorage.setItem(
+        kCustomPreference,
+        document.documentElement.classList.toggle('dark-mode')
+      );
+    });
+  }
+}
+
+function setupPickers() {
+  function closeAllPickers() {
+    for (const picker of pickers) {
+      picker.parentNode.classList.remove('expanded');
+    }
+
+    window.removeEventListener('click', closeAllPickers);
+    window.removeEventListener('keydown', onKeyDown);
+  }
+
+  function onKeyDown(e) {
+    if (e.key === 'Escape') {
+      closeAllPickers();
+    }
+  }
+
+  const pickers = document.querySelectorAll('.picker-header > a');
+
+  for (const picker of pickers) {
+    const parentNode = picker.parentNode;
+
+    picker.addEventListener('click', (e) => {
+      e.preventDefault();
+
+      /*
+        closeAllPickers as window event trigger already closed all the pickers,
+        if it already closed there is nothing else to do here
+      */
+      if (parentNode.classList.contains('expanded')) {
+        return;
+      }
+
+      /*
+        In the next frame reopen the picker if needed and also setup events
+        to close pickers if needed.
+      */
+
+      requestAnimationFrame(() => {
+        parentNode.classList.add('expanded');
+        window.addEventListener('click', closeAllPickers);
+        window.addEventListener('keydown', onKeyDown);
+      });
+    });
+  }
+}
+
+function setupStickyHeaders() {
+  const header = document.querySelector('.header');
+  let ignoreNextIntersection = false;
+
+  new IntersectionObserver(
+    ([e]) => {
+      const currentStatus = header.classList.contains('is-pinned');
+      const newStatus = e.intersectionRatio < 1;
+
+      // Same status, do nothing
+      if (currentStatus === newStatus) {
+        return;
+      } else if (ignoreNextIntersection) {
+        ignoreNextIntersection = false;
+        return;
+      }
+
+      /*
+        To avoid flickering, ignore the next changes event that is triggered
+        as the visible elements in the header change once we pin it.
+
+        The timer is reset anyway after few milliseconds.
+      */
+      ignoreNextIntersection = true;
+      setTimeout(() => {
+        ignoreNextIntersection = false;
+      }, 50);
+
+      header.classList.toggle('is-pinned', newStatus);
+    },
+    { threshold: [1] }
+  ).observe(header);
+}
+
+// Check if we have JavaScript support
+document.querySelector(':root').classList.add('has-js');
+
+// Restore user mode preferences
+setupTheme();
```

Comment:
We could start the loading of the JS file earlier and async to speed up the loading of the HTML pages. Same below. 
```suggestion
document.addEventListener('DOMContentLoaded', setupTheme, { once: true });
```

Reply:
Do you mean adding a listener for each of the functions?

Reply:
Either one listener per function, either we put everything in a single function. Your call, I doubt it makes much difference in performance, so use the pattern you find the most readable.

Reply:
Done. Does it look fine now?